### PR TITLE
update groupBy documentation

### PR DESCRIPTION
**Pull request by koperagen**
It was previously reported that moveToTop is undocumented :) Plus i want emphasize important `groupBy expr` use case for those who don't know

### CODE REVIEW

File: core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/groupBy.kt [hunk 1]
```diff
@@ -18,6 +18,15 @@ import kotlin.reflect.KProperty
 
 // region DataFrame
 
+/**
+ * Column for grouping can be created inplace
```
Comment (zaleslaw) [inline core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/groupBy.kt:22]: The comment looks like a good candidate for a NOTE: section, but not like a documentation for a function (probably you leave id with TODO: add docs and with NOTE section

File: docs/StardustDocs/snippets/org.jetbrains.kotlinx.dataframe.samples.api.Analyze.groupByMoveToTop.html [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+{ name: "<span title=\"city: String\">city</span>", children: [], rightAlign: false, values: ["Tokyo"] }, 
+{ name: "<span title=\"weight: Int\">weight</span>", children: [], rightAlign: true, values: ["<span class=\"formatted\" title=\"\"><span class=\"numbers\">68</span></span>"] }, 
+{ name: "<span title=\"isHappy: Boolean\">isHappy</span>", children: [], rightAlign: false, values: ["true"] }, 
+], id: 6, rootId: 1, totalRows: 1 } ) });
+/*-->*/
+
+/*<!--*/
+call_DataFrame(function() { DataFrame.addTable({ cols: [{ name: "<span title=\"firstName: String\">firstName</span>", children: [], rightAlign: false, values: ["Alice"] }, 
+{ name: "<span title=\"lastName: String\">lastName</span>", children: [], rightAlign: false, values: ["Wolf"] }, 
+{ name: "<span title=\"name: DataRow\">name</span>", children: [0, 1], rightAlign: false, values: ["<span class=\"formatted\" title=\"firstName: Alice\nlastName: Wolf\"><span class=\"structural\">{ </span><span class=\"structural\">firstName: </span>Alice<span class=\"structural\">, </span><span class=\"structural\">lastName: </span>Wolf<span class=\"structural\"> }</span></span>"] },
```
Comment (zaleslaw) [inline docs/StardustDocs/snippets/org.jetbrains.kotlinx.dataframe.samples.api.Analyze.groupByMoveToTop.html:555]: Do we have described somewhere a process how to generate these html files?

**Review by zaleslaw — APPROVED** [review]
Please give some clarification on my comments

**Comment by koperagen** [comment]
@Jolanrensen Have a look at the last commit please, it's related to new static table

**Pull request merged**